src/main/java/frc/robot/pathplanner/CommandChooser.java

```diff
@@ -0,0 +1,99 @@
+package frc.robot.pathplanner;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FilenameFilter;
+import java.util.HashMap;
+
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+
+import com.pathplanner.lib.commands.PathPlannerAuto;
+
+import edu.wpi.first.wpilibj.Filesystem;
+import edu.wpi.first.wpilibj.smartdashboard.SendableChooser;
+import edu.wpi.first.wpilibj2.command.Command;
+
+public class CommandChooser {
+  private HashMap<String, SendableChooser<String>> m_choosers = new HashMap<>();
+
+  public CommandChooser() {
+    try {
+      File dir = new File(Filesystem.getDeployDirectory(), "pathplanner/autos");
+      File[] autolist = dir.listFiles(new FilenameFilter() {
+        @Override
+        public boolean accept(File dir, String name) {
+          return name.endsWith(".auto");
+        }
+      });
+
+      if (autolist != null) {
+        for (var file : autolist) {
+          String folderName = getFolderForAuto(file);
+          if (folderName.isEmpty()) {
+            // Ignore autos that aren't in a folder.
+            continue;
+          }
+
+          String name = file.getName().replace(".auto", "");
+          if (!m_choosers.containsKey(folderName)) {
+            SendableChooser<String> chooser = new SendableChooser<>();
+            chooser.setDefaultOption(name, name);
+            m_choosers.put(folderName, chooser);
+          } else {
+            m_choosers.get(folderName).addOption(name, name);
+          }
+        }
+      }
+    } catch (Exception e) {
+      System.out.println("Failed to read path planner folder structure: " + e.getMessage());
```

Comment:
Unclear how to handle exceptions here, but if we let PathPlanner handle them, it will just throw runtime exceptions and halt the robot program.